Design Patterns/Examples/CSharpDesignPatterns/CSharpDesignPatterns.Prototype/Models/Cat.cs

```diff
@@ -0,0 +1,21 @@
+namespace CSharpDesignPatterns.Prototype.Models;
+
+internal class Cat : Animal
+{
+    private readonly string _species;
+
+    internal Cat (string species)
+    {
+        _species = species;
+    }
+
+    internal override Animal Clone()
+    {
+        return new Cat(_species);
```

Comment:
Well, the wider context for this set of examples is that it is for juniors - they almost certainly won't know much of anything about cloning, so all examples are designed to be reduced as far down as they can possibly be - It was playing on my mind as to whether to have more properties though, so I might go back and add some more in for a better comparison.

It is less about crafting the ideal code and more about teaching them the shape and make-up of the patterns so they can identify scenarios where they might use them. If they recognize a scenario where using a Prototype might make sense and implement it, that then means there'll be a PR where this can be a conversation.

I've avoided using records as they're less general purpose than using classes, so just another simplification - but yeah, you're right about the record clone being simpler.

P.S. Don't resolve this after reading, I'll use it to remind me about the extra props.